### ISSUE TITLE
Danger of state conflicts

### ISSUE THREAD

**Issue by peterbe** [mozilla/experimenter#1062]
I've seen a couple of these. Here's one example: https://github.com/mozilla/experimenter/blob/master/app/experimenter/experiments/forms.py#L859

It looks like this:

```python
class SomeModelForm:

    class Meta:
        model = Experiment

    def save(self, *args, **kwargs):
        experiment = super().save(*args, **kwargs)

        if some_condition:
             some_instance.save()
        if other_condition:
            SomeModel.objects.create(...)
```

So far nothing terrible has happened, that I'm aware of, in production. But I've definitely been bitten by it when dev'ing on something such that an exception is raised further down in the "meat" of the `save` method. Because the whole, default, implementation is *not* atomic, the first `save()` might work but the latter won't. What you end up with is potentially a state conflict. 

The relative risk is small because, I think, most cases of this is something like:

1. Save changes to experiment
2. Make a notification
3. Send an instruction to Celery to send an email
4. Send an instruction to Celery to update Bugzilla

In other words, many times the stuff that happens after that critical experimenter save isn't super important.

**Comment by jaredkerim** [comment]
We are undergoing major refactors for the existing bugzilla task logic which I think should address this.
